File: 600/633 - Sum of Square Numbers.java
import java.util.HashSet;

class Solution {
    public boolean judgeSquareSum(int c) {
        HashSet<Integer> set = new HashSet<Integer>();
        int i, n = (int)Math.sqrt(c);
        
        for (i = 0; i <= n; ++i) {
            set.add(i * i);
            if (set.contains(c - i * i))
                return true;
        }
        
        return false;
    }
}

File: csharp/633.sum-of-square-numbers.cs
/*
 * @lc app=leetcode id=633 lang=csharp
 *
 * [633] Sum of Square Numbers
 */

using System;

// @lc code=start
public class Solution {
    public bool JudgeSquareSum(int c) {
        int h = c / 2;
        for (int i = 0; i * i <= h; ++i) {
            int r = (int) Math.Sqrt(c - i * i);
            if (r * r + i * i == c) {
                return true;
            }
        }
        return false;
    }
}
// @lc code=end

